docs/system_overview/configuration.md

```diff
@@ -923,3 +923,52 @@ $ iadmin get_grid_configuration authentication password_max_time
+
+### Server: `server_config` Advanced Settings
+
+The `server_config.json` file for configuration of the iRODS server has a setting under `advanced_settings` called `default_number_of_transfer_threads`. This is the server's configuration of the number of threads to use by default in parallel transfers. If the client does not provide a number of threads to use for parallel transfer, this is the default value used in the other server-side configurations that will be explained in the following sections. Additionally, if `acSetNumThreads` is left in its default implementation (with "default" used for all parameters), the `default_number_of_transfer_threads` will also ensure that no client-provided number of transfer threads will exceed the value provided.
+
+The administrator may wish to disable parallel transfer threads altogether. This can be done by setting `default_number_of_transfer_threads` to 0 in the `advanced_settings`.
+
+It is recommended that careful consideration be given by the administrator when changing the default value. This will affect available resources on the configured server and could degrade performance if mis-configured. The iRODS Consortium researched the parallel transfer threads as it relates to performance and reached the default value in use today as a result. As such, it is recommended to leave the default value in place for most use cases. For more information on this topic, you can find the research paper here: [https://irods.org/uploads/2016/09/russell2016-iRODS-networking-419.pdf](https://irods.org/uploads/2016/09/russell2016-iRODS-networking-419.pdf)
+
+### How many threads will my transfer use?
+
+The following table shows the number of threads used in parallel transfers of any file which invokes parallel transfer:
+
+| case | client-provided `numThreads` (`-N`) | `default_number_of_transfer_threads` (server configuration) | threads used |
+| ---- | ---- | ---- | ---- |
+| 0 | - | 4 | 4 |
+| 1 | 3 | 4 | 3 |
+| 2 | 5 | 4 | 4 |
+
+Case 0 shows that if the client does not provide a number of parallel transfer threads to use, the default configured in the server will be used. Case 1 shows that if the client provides a number of parallel transfer threads to use, that number of threads will be used. Case 2 shows that if the client provides a number of parallel transfer threads to use that is greater than `default_number_of_transfer_threads`, the value will be set to `default_number_of_transfer_threads` so as to not exceed it. So, the configured `default_number_of_transfer_threads` in `server_config.json` actually functions as a maximum on the number of threads that will be used as well when `acSetNumThreads` is left in its default implementation.
```

Comment:
Comparing the statements explaining cases 1 and 2, it feels like case 1 needs to state something about the client's thread count being honored only because it did not exceed the default_number_of_transfer_threads.

Reply:
that's fair.

Reply:
Done